chalice/package.py

```diff
@@ -1,217 +1,285 @@
-                'Value': {'Ref': 'RestAPI'},
-            },
-            'APIHandlerName': {
-                'Value': {'Ref': 'APIHandler'},
-            },
-            'APIHandlerArn': {
-                'Value': {'Fn::GetAtt': ['APIHandler', 'Arn']}
-            },
-            'EndpointURL': {
-                'Value': {
-                    'Fn::Sub': (
-                        'https://${RestAPI}.execute-api.${AWS::Region}'
-                        # The api_gateway_stage is filled in when
-                        # the template is built.
-                        '.amazonaws.com/%s/'
-                    )
+        'Outputs': {},
+        'Resources': {},
+    }
+
+    def generate_sam_template(self, resources):
+        # type: (List[models.Model]) -> Dict[str, Any]
+        template = copy.deepcopy(self._BASE_TEMPLATE)
+        for resource in resources:
+            name = '_generate_%s' % resource.__class__.__name__.lower()
+            handler = getattr(self, name, self._default)
+            handler(resource, template)
+        return template
+
+    def _generate_scheduledevent(self, resource, template):
+        # type: (models.ScheduledEvent, Dict[str, Any]) -> None
+        function_cfn_name = to_cfn_resource_name(
+            resource.lambda_function.resource_name)
+        function_cfn = template['Resources'][function_cfn_name]
+        event_cfn_name = to_cfn_resource_name(resource.resource_name)
+        function_cfn['Properties']['Events'] = {
+            event_cfn_name: {
+                'Type': 'Schedule',
+                'Properties': {
+                    'Schedule': resource.schedule_expression,
                 }
             }
         }
-    }  # type: Dict[str, Any]
 
-    def __init__(self, swagger_generator, policy_generator):
-        # type: (SwaggerGenerator, ApplicationPolicyHandler) -> None
-        self._swagger_generator = swagger_generator
-        self._policy_generator = policy_generator
-
-    def generate_sam_template(self, config, code_uri='<placeholder>'):
-        # type: (Config, str) -> Dict[str, Any]
-        template = copy.deepcopy(self._BASE_TEMPLATE)
-        resources = {
-            'APIHandler': self._generate_serverless_function(
-                config, code_uri, 'app.app', 'api'),
-            'RestAPI': self._generate_rest_api(
-                config.chalice_app, config.api_gateway_stage),
+    def _generate_lambdafunction(self, resource, template):
+        # type: (models.LambdaFunction, Dict[str, Any]) -> None
+        resources = template['Resources']
+        cfn_name = to_cfn_resource_name(resource.resource_name)
+        resources[cfn_name] = {
+            'Type': 'AWS::Serverless::Function',
+            'Properties': {
+                'Runtime': resource.runtime,
+                'Handler': resource.handler,
+                'CodeUri': resource.deployment_package.filename,
+                'Tags': resource.tags,
+                'Timeout': resource.timeout,
```

Comment:
Some of these values used to have `if is not None` check on it. It may be good to check that these values are actually never `None` to make sure it does not affect the generated cloudformation template.

Reply:
This happens at a separate layer above.  We previously had to check at this layer because we were using the `config` object directly.  I'll double check, but I'm pretty sure it's now impossible for these values to be None now that we're using the models.